### DISCUSSION TITLE
Doesn't work with SAML Google Workspase

### DISCUSSION THREAD

**Discussion by lunahod2022** [bergie/passport-saml#690] [General]
Can't get it to work with SAML Google. This module is in Wiki.JS. What should I enter in the settings of Google itself?

**Comment by cjbarth** [comment]
Do you have a link to what you are referencing?

**Reply by lunahod2022** [reply]
I don't quite understand what you mean

**Reply by cjbarth** [reply]
You mentioned Wiki.JS. I don't know what that is.

**Reply by srd90** [reply]
@cjbarth `wiki.js` is https://github.com/requarks/wiki

@lunahod2022 be aware that current `wiki.js` has very unsecure version of `passport-saml`. At the moment `wiki.js`'s `passport-saml` instance can be configured so that authentication is effectively disabled. I.e. attacker could impersonate anyone if `wiki.js` administrator does not pay attention to `cert` parameter and such configuration error is remotely exploitable. For further information see: https://github.com/requarks/wiki/discussions/5180

**Reply by lunahod2022** [reply]
@srd90 how can i make `passport-saml` more secure? update it? Or enable forced verification of certificates?

And the question is still open. I do not understand how to properly configure authentication with Google Workspace.

**Reply by cjbarth** [reply]
That project needs to upgrade the version of `passport-saml` they are using. We are on v3 and about to release v4. There have been a ton of changes since 1.3.5.

For us to help you configure something, it would be nice to know what you've tried and what problems you're encountering.

**Reply by lunahod2022** [reply]
@cjbarth I filled in the fields "Entry point" - URL SSO, "Issuer" - google_workspace, "Certificate" in the wiki.js. In Google I filled "URL ACS" and "Object ID".
![image](https://user-images.githubusercontent.com/104645661/167125232-3db4f4f5-daf2-4ffe-9d14-2c7dff9ab1b7.png)

When I try to log in, I get various errors from Google. Now the error is "the application is not available to the user", although in the security settings it is active for everyone.

**Reply by lunahod2022** [reply]
@cjbarth @srd90 several times I tried to configure SAML in different ways, but something does not work out. I don’t understand what settings I should specify on the Google side and I’m not sure if the settings on the Wiki.JS side are entered correctly

**Comment by cjbarth** [comment]
I see that Wiki.JS has updated to v3 of `passport-saml`. Has this helped any of your issues?

**Reply by srd90** [reply]
fwiw, based on this report: https://github.com/node-saml/passport-saml/issues/521#issuecomment-848244267
`passport-saml` has **not** been "root cause" for [wiki.js](https://github.com/requarks/wiki) & google workspaces related authentication problems.
